integration/integration_test.go

```diff
@@ -1384,13 +1385,42 @@ func TestEmbedBinaryImage(t *testing.T) {
 	_, err = remote.Image(ref)
 	require.NoError(t, err, "expected image to be present after build + push")
 
-	ctr, err := runEnvbuilder(t, options{env: []string{
-		envbuilderEnv("FALLBACK_IMAGE", ref.String()),
-	}})
+	ctx := context.Background()
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		cli.Close()
+	})
+
+	// Pull the image we just built
+	rc, err := cli.ImagePull(ctx, ref.String(), image.PullOptions{})
+	require.NoError(t, err)
+	t.Cleanup(func() { _ = rc.Close() })
```

Comment:
@johnstcn why not use `defer rc.Close()` instead?

Reply:
I think that would also work here, I just prefer `t.Cleanup` inside tests.